### ISSUE TITLE
Unable to create OpenGL context

### ISSUE THREAD

**Issue by lexcass** [moonlight-stream/moonlight-chrome#273]
Ubuntu 14.04
Chrome version 59.0.3071.115
Moonlight version 0.7.3

Everything works perfectly until I start a game. The game will run on my gaming PC, and after a loading screen, my laptop (client) will get an error saying "unable to create OpenGL context."

I already set these flags in chrome://flags (as mentioned in other threads):
    - enable-nacl
    - ignore-gpu-blacklist

I also made sure that hardware acceleration was enabled in Chrome's settings. Are there any other settings or flags that would resolve this issue? Will this happen if my system's specs are too low?

**Comment by Jorys-Paulin** [comment]
- How about the client specs?
- Have you installed OpenGL on your client, if on Linux?

**Comment by lexcass** [comment]
Client CPU: Intel(R) Core(TM) i3-4030U CPU @ 1.90GHz (x86_64)
Client Mem: 4GB Memory (DDR3L)
Client GPU: Intel HD Graphics 4400
    product: Haswell-ULT Integrated Graphics Controller
    width: 64 bits
    clock: 33MHz

I already had OpenGL, but I reinstalled it just to be safe. The problem persisted. The above OS and chrome versions are for the client (laptop), and the OpenGL version below is from running glxinfo.

OpenGL version string: 3.0 Mesa 10.1.3

**Comment by Jorys-Paulin** [comment]
And the client OS is?

**Comment by Jorys-Paulin** [comment]
The error appends at https://github.com/moonlight-stream/moonlight-chrome/blob/master/viddec.cpp#L93, wich means that Moonlight can't initialize the rendering surface, so that's the actual element that displays your screen. It surely is an OpenGL error, but I'm not expert in OpenGL or C/C++

**Comment by lexcass** [comment]
**Client OS**
Ubuntu 14.04
Chrome version 59.0.3071.115
Moonlight version 0.7.3

Thank you very much. I appreciate your help. I will see what I can decipher from the code, and if I find anything, I will post the solution here.

**Comment by Jorys-Paulin** [comment]
Have a look at that: #243
@jacobmix asked him to check [chrome://gpu/](chrome://gpu/) for anything GPU-related

**Comment by Jorys-Paulin** [comment]
Hey @lexcass, do you have resolved your problem?

**Comment by lexcass** [comment]
Unfortunately, no. The last thing I tried was installing OpenGL ES since it was included in the source file you linked.

**Comment by Jorys-Paulin** [comment]
@lexcass Since you're on Ubuntu, do you have OpenGL installed and all graphics librairies? Just to be sure...

**Comment by Jorys-Paulin** [comment]
@lexcass I'll test on my Ubuntu Virtual machine, see what I got. I'll come back to you then :)

**Comment by Jorys-Paulin** [comment]
Does other WebGL/OpenGL applications work in chrome? Does it require an activation of some sort?

**Comment by lexcass** [comment]
Yes, other WebGL/OpenGL applications work in Chrome. I used the same flags with these applications as I did with Moonlight, and Moonlight is still unable to create an OpenGL context.

The applications didn't run very well (just a few FPS), so my hardware may not be fully equipped to handle WebGL. However, I believe it would at least create an OpenGL context as the others did.

**Comment by lexcass** [comment]
If I run Moonlight from the console I get this error:

[6856:6891:0721/095711.809602:ERROR:browser_gpu_channel_host_factory.cc(103)] Failed to launch GPU process.
Created new window in existing browser session.

**Comment by Jorys-Paulin** [comment]
Are you running an official build or an unpacked extension? Is the menu working?

**Comment by lexcass** [comment]
I  followed the instructions from this page:

https://github.com/moonlight-stream/moonlight-chrome/releases

The menu is working, and the application connects to my gaming computer and displays my games.

**Comment by Jorys-Paulin** [comment]
So the problem is fixed?

**Comment by lexcass** [comment]
No, sorry. I should have clarified that. The menu and games display, and selecting a game runs the game on my gaming computer. However on the client, it still tells me "unable to create OpenGL context."

**Comment by Jorys-Paulin** [comment]
I think we'll need more help, I can't quite dive into the code to try to catch the problem because I'm not a C++ developer, but maybe @cgutman can help us on that.

**Comment by Inthrall** [comment]
I'm getting this same issue now, is there any more information I can provide here to help?
